axum/src/routing/mod.rs

```diff
@@ -419,24 +415,8 @@ where
 
         #[cfg(feature = "matched-path")]
         if let Some(matched_path) = self.node.route_id_to_path.get(&id) {
-            use crate::extract::MatchedPath;
-
-            let matched_path = if let Some(previous) = req.extensions_mut().get::<MatchedPath>() {
-                // a previous `MatchedPath` might exist if we're inside a nested Router
-                let previous = if let Some(previous) =
-                    previous.as_str().strip_suffix(NEST_TAIL_PARAM_CAPTURE)
-                {
-                    previous
-                } else {
-                    previous.as_str()
-                };
-
-                let matched_path = format!("{}{}", previous, matched_path);
-                matched_path.into()
-            } else {
-                Arc::clone(matched_path)
-            };
-            req.extensions_mut().insert(MatchedPath(matched_path));
+            req.extensions_mut()
+                .insert(MatchedPath(Arc::clone(matched_path)));
```

Comment:
This change means that nested opaque services will only see the matched path from their point of view.

i.e. with

```rust
Router::new().nest_service(
    "/a",
    Router::new().route("/:foo", ...)
)
```

the inner router will see the matched path as `/:foo` and _not_ `/a/:foo`. That is consistent with how `Uri` works and why we have `OriginalUri`.

Reply:
Ah wait a sec. I reverted this in https://github.com/tokio-rs/axum/pull/1086/commits/657a0893b51f581b0090747cfdbe95070d126483 because I assumed it was a bug. Had forgotten about this comment. Actually not sure which behavior is better. I'll think about it!

Reply:
I can't quite decide what the best behavior is here so think we should keep it the way it worked for 0.5.